### PR TITLE
[to discuss] exec blocks without ObjectDb: Mutation over TxDb

### PR DESCRIPTION
**Pull request by AskAlexSharov**
- Usage of TxDb means don't open read transaction every time you wanna read something from db. 
- On fast disk it speeding genesis sync 21hour -> 17hours
- But it a bit increases complexity of stage_exec - now there is db.Begin, then tx.NewBatch, then both of them need periodical commit
- I need your advise - maybe we can simplify code somehow?

### CODE REVIEW

**Comment by AlexeyAkhunov** [comment]
I don't see it as a big increase in complexity at all. Will test the full sync with it myself soon

File: eth/stagedsync/stage_execute.go [hunk 1]
```diff
@@ -132,6 +147,9 @@ func SpawnExecuteBlocksStage(s *StageState, stateDB ethdb.Database, chainConfig
 	if _, err := batch.Commit(); err != nil {
 		return fmt.Errorf("sync Execute: failed to write batch commit: %v", err)
 	}
+	if _, err = tx.Commit(); err != nil {
```
Comment (mandrigin) [inline eth/stagedsync/stage_execute.go:150]: what will happen when we Commit and the defer calls Rollback?

Comment (AskAlexSharov) [reply eth/stagedsync/stage_execute.go:150]: Commit sets internal transaction object to nil, then rollback do something only if that object is not nil. So, rollback after commit must not do anything. Same for double-rollback.

Comment (AskAlexSharov) [reply eth/stagedsync/stage_execute.go:150]: With Tx need to be more strict than with mutation: if tx not closed well - it can cause deadlock or leak.

**Pull request merged**